### DISCUSSION TITLE
How to Generating code signing DER files with only access to the hash of the image file?

### DISCUSSION THREAD

**Discussion by berlin4apk** [ipxe/ipxe#485] [Q&A] [answered]
How can i Generating code signing DER signature files with only access to the fingerprint of the image files?

If I follow the iPXE cryptography guide https://ipxe.org/crypto 
I must have the file to sign locally, witch is a pain if I wish to sign a lot of big ISO / squasfs files.

So how can I "convert" a sha256 hash list or gpg signature to a detached code signing DER file?

As form my interpretation the signature file contains only the image fingerprint and the fingerprint 
Certificates used for code signing. (witch must include the digitalSignature key usage extension and the codeSigning extended key usage extension.)

**Comment by mcb30** [comment, marked as answer]
This is a limitation of the `openssl cms` tool rather than iPXE.  You can find what seems to be relevant openssl issue at https://github.com/openssl/openssl/issues/14686

**Comment by berlin4apk** [comment]
Thank you for the issue Link!

A workaround would be to combination of a sha256sum command
And a CMS singed iPXE script directly in iPXE

But I don't know how big would be the interpenetration manpower and the increase of the iPXE binarys

@mcb30 can you make a guess if it's makes sense to go this "way"? Thank you for your time

**Reply by mcb30** [reply]
This is not a workaround that I'm prepared to accept into iPXE, sorry.  Allowing images to be marked as trusted without requiring a full signature check opens up too many possible ways for people to create scripts with security holes.

I would suggest contributing to the openssl issue, since that provides the proper fix for your problem.

**Reply by berlin4apk** [reply]
Thank you for the feedback about a workaround,
I will take a look about the openssl issue.
